book/embeds.md

```diff
@@ -145,104 +183,133 @@ def request(url, top_level_url, payload=None):
     return headers, body
 ```
 
-Now define a method that decodes a response body that we know is an image
-(even if we don't know its format).^[Interestingly, to make it work for our toy
-browser we don't need to consult `content-type`. That's because Pillow already
-auto-detects the image format by peeking at the first few bytes of the binary
-data, which varies for each image format.] First, reinterpret
-the image "file" as a `BytesIO` object and pass it to Pillow. Then convert
-it to RGBA (the same RGBA as in
-[Chapter 11](/visual-effects.html#sdl-creates-the-window)), call `tobytes`
-(which performs the decode and puts the result in a raw byte
-array[^maybe-decode]), and wrap the result in a Skia `Image` object.
-
-[^maybe-decode]: Maybe. As with Skia, Pillow tries to be lazy about when to
-decode, so probably the decode happens at this time. But there is nothing
-in the Pillow API that requires it to decode right then, rather than say in
-the `open` call. For our toy browser it doesn't matter very much, but in a
-real browser the timing of a decode is important for performance. That's also
-why there is an [HTML API][html-image-decode] to control decoding.
+Interestingly, we don't consult `Content-Type` to decode images.
+That's because Pillow auto-detects the image format by peeking at the
+first few bytes of the binary data. Image formats typically include
+"magic bytes" that identify the image format at the beginning.
 
-[html-image-decode]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/decoding
+:::
 
-``` {.python expected=False}
-def decode_image(image_bytes):
-    picture_stream = io.BytesIO(image_bytes)
-    pil_image = PIL.Image.open(picture_stream)
-    if pil_image.mode == "RGBA":
-        pil_image_bytes = pil_image.tobytes()
-    else:
-        pil_image_bytes = pil_image.convert("RGBA").tobytes()
+Before we can use a downloaded image, we need to _decode_ it. That's
```

Comment:
```suggestion
After we can use a downloaded image, we need to _decode_ it. That's
```

Reply:
Don't think the proposal is correct but will rephrase.